### ISSUE TITLE
[STORY]: Create new email notification for 'Mark as Interested'

### ISSUE THREAD

**Issue by ekhoffman** [usdigitalresponse/usdr-gost#2962]
### Full User Story

As an organization, we would like to foster more collaboration between our users to allow for the sharing of knowledge and resources to empower them to more efficiently and effectively apply for the grants discovered on our platform.

As a grantseeker, I want to be able to easily share grants I’ve discovered with my team.

As a grantseeker, I want to be able to easily find collaborators who may be able to help me when preparing and applying for grants.

As a grantseeker, I want to be able to quickly see what grants my team members are interested in.

As an admin, I want to be able to easily surface and resolve issues and blockers my team is running into.

### Why is this issue important?

In order to help build the collaborative spirit we're going for, we need to have a prompt to pull people back to the platform when people they may be interested in supporting or receiving support from mark interest in the same grant. This will help bring users back to the platform in a timely manner to collaborate on previously marked grants.

### Action Plan

- [ ] Initial design drafted
- [ ] Initial acceptance criteria drafted
- [ ] High level technical approach & feature flag use determined
- [ ] Designs & Acceptance criteria finalized
- [ ] Eng task breakout complete
- [ ] Assessment of whether partner training & communication actions are needed
- [ ] Acceptance criteria testing completed & passed in staging
- [ ] Feature flags enabled in production
- [ ] Feature flags are removed once the feature is deemed stable

### Engineering Tasks

Add links to engineering tasks

### Acceptance Criteria

1. Create new email notification for 'Mark as Interested.'
    - Trigger: New user in the organization marks interest on a grant.
    - Recipients: Any other user in that organization who has previously marked interest (and not opted out of emails).
    - Email Subject: A new collaborator is interested in {{Grant Name}}
    - Email Title: New Collaborator
    - Email Body: 
    {{User Name}} of {{Team Name}} has marked interest in {{Grant Name}}. 
    [If Interest Note included] They included the following note: {{Interest Note}}
    Grant Detail Box

**My Profile**
1. Add new entry to Email Notifications.
    - Name: Mark as Interested
    - Description: Send me notifications when collaborators mark interest on a grant.

**Comment by samserif23** [comment]
![New Collaborator Email -- Complex Layout](https://github.com/usdigitalresponse/usdr-gost/assets/157177162/74ed1a67-8bf1-4d33-b53f-ca57d4a0d1bf)
![New Collaborator Email -- Simple Layout](https://github.com/usdigitalresponse/usdr-gost/assets/157177162/b6e586b0-791c-4188-b864-c5b00a044d5a)

**Comment by samserif23** [comment]
Two leading designs thus far, one exploring pulling in the user avatar + comment visual component into the email

**Comment by samserif23** [comment]
Additional comment: View My Grants button links to interested tab

**Comment by TylerHendrickson** [comment]
Drafting some scratch for implementing these notes as a digest...

### Querying for grant activity

The following query returns both new follows and note revision content for grants since a given `:sinceDate`, limited to a given `tenan.id` identified by `:organizationId`. In this example, the rows are ordered by grant ID (to keep rows pertaining to the same grant together), then new follower activity (oldest first), then new note revision content (oldest first).
```sql
SELECT
    g.grant_id AS grant_id,
    g.title AS grant_title,
    u.id AS user_id,
    u.name AS user_name,
    u.email AS user_email,
    a.name AS agency_name,
    activity.activity_at,
    activity.activity_type,
    activity.text_content AS activity_text_content
FROM grants g
JOIN (
    SELECT
        gf.id,
        gf.grant_id,
        gf.user_id,
        gf.created_at AS activity_at,
        'follow' AS activity_type,
        null AS text_content
    FROM grant_followers gf
    UNION ALL
    SELECT
        rev.id,
        gn.grant_id,
        gn.user_id,
        rev.created_at AS activity_at,
        'note' AS activity_type,
        rev.text AS text_content
    FROM grant_notes gn
    LEFT JOIN LATERAL (
        SELECT
            r.id,
            r.grant_note_id,
            r.created_at,
            r.text
        FROM grant_notes_revisions r
        WHERE r.grant_note_id = gn.id
        ORDER BY r.created_at DESC
        LIMIT 1
    ) AS rev ON rev.grant_note_id = gn.id
) activity ON activity.grant_id = g.grant_id
JOIN users u ON u.id = activity.user_id
JOIN agencies a ON a.id = u.agency_id
JOIN tenants o ON o.id = u.tenant_id
WHERE 
    o.id = :organizationId
    AND activity.activity_at >= :sinceDate
ORDER BY
    g.grant_id DESC,
    -- Followers first, then notes (remove for grant activity in chronological order, regardless of type):
    array_position(array['follow', 'note'], activity.activity_type) ASC,
    activity.activity_at ASC;
```

Querying in this manner allows for the result set to be iterated over in a manner that can be used to progressively construct the contents of a digest email. Since rows with the same grant ID appear consecutively (and within those, rows representing the same activity type appear consecutively, ordered by oldest-first), the body of an iterating loop can watch for changing `grant_id` and `activity_type` values to determine when to construct a new digest section for a grant and a new sub-section pertaining to activity type within that grant.

Example result set:
| grant_id | grant_title                        | user_id | user_name        | user_email           | agency_name       | activity_at                   | activity_type | activity_text_content                         |
|----------|------------------------------------|---------|------------------|----------------------|-------------------|-------------------------------|---------------|-----------------------------------------------|
| 987654   | This grant has a lot of activity   | 5       | Jennifer Coleman | jcoleman@example.gov | Climate Office    | 2024-09-19 09:24:07.531172+00 | follow        | `null`                                        |
| 987654   | This grant has a lot of activity   | 2       | Ray Welch        | rwelch@example.gov   | Dept of Education | 2024-09-19 15:20:37.545297+00 | follow        | `null`                                        |
| 987654   | This grant has a lot of activity   | 5       | Jennifer Coleman | jcoleman@example.gov | Climate Office    | 2024-09-19 09:25:53.746174+00 | note          | First note of the day.                        |
| 987654   | This grant has a lot of activity   | 4       | Ken Ingram       | kingram@example.gov  | Public Works      | 2024-09-19 10:01:26.897312+00 | note          | This is the second note.                      |
| 987654   | This grant has a lot of activity   | 3       | Traci Tran       | ttran@example.gov    | Climate Office    | 2024-09-19 14:21:53.306255+00 | note          | Last note for today!                          |
| 246813   | A grant with a single new follower | 4       | Ken Ingram       | kingram@example.gov  | Public Works      | 2024-09-19 10:45:29.726974+00 | follow        | `null`                                        |
| 123456   | Two new followers and one new note | 3       | Traci Tran       | ttran@example.gov    | Climate Office    | 2024-09-19 15:50:42.655014+00 | follow        | `null`                                        |
| 123456   | Two new followers and one new note | 5       | Jennifer Coleman | jcoleman@example.gov | Climate Office    | 2024-09-19 16:22:08.934724+00 | follow        | `null`                                        |
| 123456   | Two new followers and one new note | 5       | Jennifer Coleman | jcoleman@example.gov | Climate Office    | 2024-09-19 16:22:09.468254+00 | note          | This is the second grant I've followed today. |

Note that this query can also make use of aggregate functions such as `json_agg()` along with `GROUP BY grant.grant_id, grant.title` so that each row represents an individual grant and provides a JSON column of aggregated activity data for that grant. Doing so can potentially simplify iteration on the JS side (although the tradeoff in query complexity may not be worth it).

**Comment by greg-adams** [comment]
@ClaireValdivia Just want to confirm whether or not the email digest a user receives will contain their *own* follow/note activity (assuming it is not the only activity for that day)

**Comment by ClaireValdivia** [comment]
@greg-adams no, the user should not see their own follow/note activity in the digest.
